includes/wc-template-functions.php

```diff
@@ -1966,7 +1966,7 @@ function woocommerce_form_field( $key, $args, $value = null ) {
 
 					$field .= '<input type="hidden" class="hidden" name="' . esc_attr( $key ) . '" id="' . esc_attr( $args['id'] ) . '" value="" ' . implode( ' ', $custom_attributes ) . ' placeholder="' . esc_attr( $args['placeholder'] ) . '" />';
 
-				} elseif ( is_array( $states ) ) {
+				} elseif ( is_array(!empty( $states )) ) {
```

Comment:
The CSS look good, but I can't merge this pull request until you remove this change. 

We do one pull request per issue, and this line doesn't have anything to do with the CSS changes. Feel free to open a separate PR with this change if there is a good reason for it, but I can tell you right now that it has some logic problems in it.